### PR TITLE
Winapi

### PR DESCRIPTION
**Pull request by upside2**
Update Windows API calls.
1 - Refactor to break out each Windows library into a separate file.
2 - Fix a few calls to properly return void (notably, ntdll.security_check_cookie).
3 - Add libraries for msvcr* for versions of MS Visual Studio from 2005 thru 2013.  Only the most frequently used calls are included, providing coverage for an estimated 80% of calls seen in binary executables.

### CODE REVIEW

**Comment by upside2** [comment]
Update Windows API calls.
1 - Refactor to break out each Windows library into a separate file.
2 - Fix a few calls to properly return void (notably, ntdll.security_check_cookie).
3 - Add libraries for msvcr* for versions of MS Visual Studio from 2005 thru 2013. Only the most frequently used calls are included, providing coverage for an estimated 80% of calls seen in binary executables.

File: envi/__init__.py [hunk 1]
```diff
@@ -1186,21 +1186,23 @@ def executeCall(self, emu, va, args=None, ra=None):
         self.setupCall(emu, args=args, ra=ra)
         emu.setProgramCounter(va)
 
-    def execCallReturn(self, emu, value, argc):
+    def execCallReturn(self, emu, value, argc, apiReturnsVoid=False):
```
Comment (atlas0fd00m) [inline envi/__init__.py]: please keep the pull requests separate.  this PR has some good value, but it will be hindered until the other one gets sorted out.  if you extricate the other PR from this one, it's more likely to get merged.

**Comment by atlas0fd00m** [comment]
still waiting on the decoupling of the apiReturnsVoid() PR form this one.  please do this, i want this one! ;)

File: envi/__init__.py (hunk 1 above)
Comment (upside2) [reply envi/__init__.py]: OK, it's backed out.  This PR is just the Windows API update.

File: vivisect/vamp/msvc/__init__.py [hunk 2]
```diff
@@ -1,32 +1,78 @@
...
+# Perhaps never used.
+_sigD2 = ('8bff558bec5151a300000000890d00000000891500000000891d00000000893500000000893d000000008c15000000008c0d000000008c1d000000008c05000000008c25000000008c2d000000009c',
+          'ffffffffffffffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ffff00000000ff',
+          'ntdll.report_gsfailure')
+
+sigs = [ _sigA1, _sigA2, _sigA3, _sigA4, _sigA5, _sigA6,
+         _sigB1, _sigB2, _sigB3,
+         _sigC1, _sigC2, _sigC3, _sigC4,
+         _sigD1, _sigD2 ]
+
```
Comment (atlas0fd00m) [inline vivisect/vamp/msvc/__init__.py:75]: can we get this moved into a separate PR (or merge master into this and resolve conflicts)?  this does still have value for VAMP.

thanks!
@

Comment (upside2) [reply vivisect/vamp/msvc/__init__.py:75]: Actually, this is already in the master branch.  It looks different because I rewrote it to match the existing style, but the logic is the same.

**Comment by upside2** [comment]
closing
